pkg/runtime/cache/accounts.go

```diff
@@ -0,0 +1,125 @@
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cache
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	informersv1 "k8s.io/client-go/informers/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	k8scache "k8s.io/client-go/tools/cache"
+)
+
+const (
+	// ACKRoleAccountMap is the name of the configmap map object storing
+	// all the AWS Account IDs associated with their AWS Role ARNs.
+	ACKRoleAccountMap = "ack-role-account-map"
+)
+
+// AccountCache is responsible for caching the CARM configmap
+// data. It is listening to all the events related to the CARM map and
+// make the changes accordingly.
+type AccountCache struct {
+	sync.RWMutex
+
+	log logr.Logger
+	// ConfigMap informer
+	informer k8scache.SharedInformer
+	roleARNs map[string]string
+}
+
+// NewAccountCache makes a new AccountCache from a client.Interface
+// and a logr.Logger
+func NewAccountCache(kc clientset.Interface, log logr.Logger) *AccountCache {
+	sharedInformer := informersv1.NewConfigMapInformer(
+		kc,
+		currentNamespace,
+		informerResyncPeriod,
+		cache.Indexers{},
+	)
+	return &AccountCache{
+		informer: sharedInformer,
+		log:      log.WithName("AccountCache"),
+		roleARNs: make(map[string]string),
+	}
+}
+
+// resourceMatchACKRoleAccountConfigMap verifies if a resource is
+// the CARM configmap. It verifies the name, namespace and object type.
+func resourceMatchACKRoleAccountsConfigMap(raw interface{}) bool {
+	object, ok := raw.(*corev1.ConfigMap)
+	return ok && object.ObjectMeta.Namespace == currentNamespace
+}
+
+// Run adds event handler functions to the SharedInformer and
+// runs the informer to begin processing items.
+func (c *AccountCache) Run(stopCh <-chan struct{}) {
+	c.informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			if resourceMatchACKRoleAccountsConfigMap(obj) {
+				object := obj.(*corev1.ConfigMap).DeepCopy()
+				c.log.Info("CARM configmap has been created")
+				c.updateAccountRoleData(object.Data)
+				c.log.Info("cached CARM config map")
+			}
+		},
+		UpdateFunc: func(old, new interface{}) {
+			if resourceMatchACKRoleAccountsConfigMap(new) {
+				object := new.(*corev1.ConfigMap).DeepCopy()
+				c.log.Info("CARM configmap has been updated")
+				//TODO(a-hilaly): compare data checksum before updating the cache
+				c.updateAccountRoleData(object.Data)
+				c.log.Info("cached CARM configmap")
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			if resourceMatchACKRoleAccountsConfigMap(obj) {
+				c.log.Info("CARM configmap has been deleted")
+				newMap := make(map[string]string)
+				c.updateAccountRoleData(newMap)
+				c.log.Info("cleaned up cache")
+			}
+		},
+	})
+
+	go c.informer.Run(stopCh)
+}
+
+// GetByKey returns the account role ARN from the cached configmap
+func (c *AccountCache) GetByKey(key string) (string, error) {
```

Comment:
Maybe call this `GetRoleARN()`? And change the variable name "key" to "accountID"